File: solutions/scala/bob/1/src/main/scala/Bob.scala
object Bob {

  def response(statement: String): String = {
    val cleanStatement = statement.trim
    if (cleanStatement == "") {
      "Fine. Be that way!"
    }
    else {
      val wasYelled           = isYelled(cleanStatement)
      val wasQuestion         = isQuestion(cleanStatement)
      val wasYelledQuestioned = wasYelled && wasQuestion
      cleanStatement match {
        case _ if wasYelledQuestioned => "Calm down, I know what I'm doing!"
        case _ if wasYelled           => "Whoa, chill out!"
        case _ if wasQuestion         => "Sure."
        case _                        => "Whatever."
      }
    }
  }

  private def isYelled(statement: String): Boolean =
    statement.exists(_.isLetter) &&
    statement == statement.toUpperCase

  private def isQuestion(statement: String): Boolean =
    statement.last == '?'
}

File: solutions/csharp/bob/1/Bob.cs
using System.Linq;

public static class Bob
{
	public static string Response(string statement)
	{
		statement = statement.Trim();
		if (string.IsNullOrWhiteSpace(statement))
		{
			return "Fine. Be that way!";
		}

		bool wasYelled = IsYelled(statement);
		bool wasQuestion = IsQuestion(statement);
		bool wasYelledQuestion = wasYelled && wasQuestion;

		return statement switch
		{
			_ when wasYelledQuestion
				=> "Calm down, I know what I'm doing!",

			_ when wasYelled
				=> "Whoa, chill out!",

			_ when wasQuestion
				=> "Sure.",

			_ => "Whatever.",
		};
	}

	private static bool IsYelled(string statement)
		=> statement.Any(char.IsLetter)
			&& statement == statement.ToUpperInvariant();

	private static bool IsQuestion(string statement)
		=> '?' == statement.Last();
}